realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -136,6 +146,403 @@ public boolean load() {
+        Class<?> valueClass = value.getClass();
+        if (valueClass == Boolean.class) {
+            setBoolean(fieldName, (Boolean) value);
+        } else if (valueClass == Short.class) {
+            setShort(fieldName, (Short) value);
+        } else if (valueClass == Integer.class) {
+            setInt(fieldName, (Integer) value);
+        } else if (valueClass == Long.class) {
+            setLong(fieldName, (Long) value);
+        } else if (valueClass == Byte.class) {
+            setByte(fieldName, (Byte) value);
+        } else if (valueClass == Float.class) {
+            setFloat(fieldName, (Float) value);
+        } else if (valueClass == Double.class) {
+            setDouble(fieldName, (Double) value);
+        } else if (valueClass == String.class) {
+            //noinspection ConstantConditions
+            setString(fieldName, (String) value);
+        } else if (value instanceof Date) {
+            setDate(fieldName, (Date) value);
+        } else if (value instanceof byte[]) {
+            setBlob(fieldName, (byte[]) value);
+        } else if (value instanceof RealmModel) {
+            setObject(fieldName, (RealmModel) value);
+        } else if (valueClass == RealmList.class) {
+            RealmList<?> list = (RealmList<?>) value;
+            setList(fieldName, list);
+        } else {
+            throw new IllegalArgumentException("Value is of an type not supported: " + value.getClass());
+        }
+    }
+
+    /**
+     * Sets the value to {@code null} for the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @throws IllegalArgumentException if field name doesn't exist or is a primary key property.
+     * @throws IllegalStateException if the field cannot hold {@code null} values.
+     */
+    public void setNull(String fieldName) {
+        checkNonEmptyFieldName(fieldName);
+        realm.checkIfValid();
+        osResults.setNull(fieldName);
+    }
+
+    /**
+     * Sets the {@code boolean} value of the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist, is a primary key property or isn't a boolean field.
+     */
+    public void setBoolean(String fieldName, boolean value) {
+        checkNonEmptyFieldName(fieldName);
+        fieldName = mapFieldNameToInternalName(fieldName);
+        checkType(fieldName, RealmFieldType.BOOLEAN);
+        realm.checkIfValid();
+        osResults.setBoolean(fieldName, value);
+    }
+
+    /**
+     * Sets the {@code byte} value of the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist, is a primary key property or isn't an integer field.
```

Comment:
isn't a byte value _user doesn't know the underlying implementation_ (same comment for below)